### PR TITLE
feat: add pagination to models endpoint and improve files pagination docs

### LINKED ISSUE #82: Add pagination to models endpoint
## Description
The models listing endpoint lacks pagination support, which could become a performance issue with many models.

## Location
`dwctl/src/api/models/deployments.rs:13`

## Context
```rust
/// Query parameters for listing deployed models
/// TODO: Pagination
#[derive(Debug, Deserialize, IntoParams, ToSchema)]
pub struct ListModelsQuery {
    pub endpoint: Option<InferenceEndpointId>,
    pub include: Option<String>,
    pub deleted: Option<bool>,
    pub inactive: Option<bool>,
    pub accessible: Option<bool>,
}
```

## Proposed Solution
Add pagination parameters similar to the files endpoint:
- `limit` - number of items per page
- `offset` or cursor-based pagination
- Return pagination metadata (total, has_more, etc.)

## Impact
- Better performance with large model catalogs
- Consistent pagination across API endpoints
- Improved API scalability

### LINKED ISSUE #83: Implement offset-based pagination or remove skip field
## Description
The `ListFilesQuery` has a `skip` field for pagination, but it's marked as dead code and not actually implemented in the handler.

## Location
`dwctl/src/api/models/files.rs:13`

## Context
```rust
#[derive(Debug, Deserialize, IntoParams, ToSchema)]
pub struct ListFilesQuery {
    pub limit: Option<i64>,

    #[allow(dead_code)] // TODO: Implement offset-based pagination or remove this field
    pub skip: Option<i64>,

    pub order: String,
}
```

## Proposed Solution
Either:
1. Implement the skip/offset pagination in the files handler
2. Remove the field if not needed
3. Document why it's there but unused (if there's a valid reason)

## Impact
- Removes dead code
- Either adds pagination feature or cleans up API surface
- Makes API documentation accurate

### PR DESCRIPTION
**Pull request by sejori**
## Overview
This PR addresses issues #82 and #83 by implementing pagination for the models endpoint and improving documentation for the files endpoint's existing cursor-based pagination.

### Issue #82: Add pagination to models endpoint
- ✅ Added `limit` parameter (default: 10, max: 100) for controlling number of results
- ✅ Added `skip` parameter (default: 0) for skipping results
- ✅ Added `search` parameter for filtering models by alias or model_name
- ✅ Response now uses `PaginatedResponse<T>` wrapper with `data`, `total_count`, `skip`, and `limit` fields
- ✅ Created reusable `PaginationQuery` struct in `api/models/pagination.rs` for consistent pagination across endpoints
- ✅ Updated OpenAPI documentation with new parameters
- ✅ Added comprehensive test suite (`test_models_pagination`)

### Issue #83: Cursor-based pagination for files endpoint
- ✅ Files endpoint already had proper cursor-based pagination with `after` parameter
- ✅ Enhanced documentation to clarify OpenAI-compatible pagination
- ✅ Improved OpenAPI docs explaining how to use `last_id` from response

### Models Endpoint Pagination
- Uses offset-based pagination with `skip` and `limit` parameters
- Default limit of 10 with a maximum cap of 100 for performance
- Returns `PaginatedResponse` with:
  - `data`: Array of model items
  - `total_count`: Total matching items (for calculating pages)
  - `skip`: Number of items skipped
  - `limit`: Items per page
- Maintains backward compatibility (pagination params are optional)
- Works with all existing filters (endpoint, accessible, deleted, inactive, include, search)

### Reusable Pagination Types
- `PaginationQuery`: Input DTO for pagination parameters (`skip`, `limit`, `search`)
- `PaginatedResponse<T>`: Generic response wrapper for paginated data
- Located in `dwctl/src/api/models/pagination.rs` for use across all endpoints

### Files Endpoint
- Already implemented cursor-based pagination following OpenAI conventions
- Returns `first_id`, `last_id`, and `has_more` in response
- Fetches N+1 items to determine if more results exist

## Testing
- ✅ All tests passing
- ✅ Pagination tests validate:
  - Multiple pages with different limit/skip combinations
  - No duplicate results across pages
  - Correct handling of edge cases (skip beyond results)
  - Default behavior without pagination params
  - Total count accuracy

### Models with pagination:
```bash
# Get first page (10 items by default)
GET /admin/api/v1/models

# Get first page with custom limit
GET /admin/api/v1/models?limit=20&skip=0

# Get second page
GET /admin/api/v1/models?limit=20&skip=20

# Search with pagination
GET /admin/api/v1/models?search=gpt&limit=10
```

### Response format:
```json
{
  "data": [...],
  "total_count": 42,
  "skip": 0,
  "limit": 10
}
```

### Files with cursor pagination:
```bash
# Get first page
GET /ai/v1/files?limit=20

# Get next page using last_id from response
GET /ai/v1/files?limit=20&after={last_id}
```

Closes #82
Closes #83

### CODE REVIEW

File: dwctl/src/api/models/batches.rs [hunk 1]
```diff
@@ -144,14 +144,13 @@ pub enum ListObjectType {
...
 pub struct ListBatchesQuery {
-    /// A cursor for pagination (after this batch ID)
-    #[serde(rename = "after")]
-    pub after: Option<String>,
-
-    /// Maximum number of batches to return (1-100, default 20)
-    #[param(default = 20, minimum = 1, maximum = 100)]
-    pub limit: Option<i64>,
+    /// Pagination parameters
+    #[serde(flatten)]
```
[the remark this answers is not in the document]
Comment (sejori) [reply dwctl/src/api/models/batches.rs]: Honestly it was pretty smooth - I was expecting to have to wrestle with rust a bit more but the macros seemed to just work. I need to make sure the utopia does render everything correctly though...
